homework10/homework10_3.py

```diff
@@ -0,0 +1,41 @@
+"""Homework10_3:Functions. Writing and implementing programs."""
+
+
+# Подсчет количества букв
+# На вход подается строка, например, "cccbba" результат работы
+# программы - строка “c3b2a"
+
+def count_letters(given_string: str) -> str:
+    """
+     Returns a string with
+     unique characters and their quantity.
+     """
+    result_string = ''
+    letters = {}
+
+    def count_and_remove(cut_string: str) -> str:
+        nonlocal result_string, letters
+        for idx, letter in enumerate(cut_string):
+            if letter not in letters:
+                letters[letter] = 1
+            else:
+                letters[letter] += 1
+            if len(cut_string) == 1 or len(set(cut_string)) == 1:
+                result_string += (letter + str(len(cut_string)))
+                return result_string.replace('1', '')
+            if cut_string[idx] != cut_string[idx + 1]:
+                result_string += (letter + str(letters.get(letter)))
+                cut_string = cut_string[idx + 1:]
+                letters = {}
+                return count_and_remove(cut_string)
+        # to pass pylint R1710
+        return ''
+
+    return count_and_remove(given_string)
```

Comment:
Not so complex solution:
```suggestion
    result = ""
    current_char = s[0]
    count = 1

    for i in range(1, len(s)):
        if s[i] == current_char:
            count += 1
        else:
            result += current_char + (str(count) if count > 1 else "")
            current_char = s[i]
            count = 1

    result += current_char + (str(count) if count > 1 else "")

    return result
```

Reply:
thank you so much!!